src/core/src/sys/molecules.rs

```diff
@@ -154,10 +28,8 @@ pub struct Molecule {
     /// the connectivity between the particles `i + self.first` and
     /// `j + self.first`
     connections: Array2<Connectivity>,
-    /// Index of the first atom in this molecule.
-    first: usize,
-    /// Index of the last (included) atom in this molecule.
-    last: usize,
+    /// Range of atomic indexes in this molecule.
+    atoms: Range<usize>,
```

Comment:
Maybe rename this to something like `particles_range`? I think `particles` would also work but could lead to confusion.